### PR TITLE
feat: show type badge in DataGrid header

### PR DESCRIPTION
**Pull request by keita-determined**
- [ET-792]
- Show type badge in DataGrid header

https://github.com/user-attachments/assets/8a19a9da-1385-4d86-94d0-c02ce955e2c3

[ET-792]: https://hpe-aiatscale.atlassian.net/browse/ET-792?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/kit/DataGrid/custom-renderers/utils.ts [hunk 1]
```diff
@@ -105,7 +107,30 @@ export function drawTextWithEllipsis(
...
+
+  ctx.beginPath();
+  ctx.font = tagFont;
+  ctx.fillStyle = backgroundColor;
+  ctx.strokeStyle = backgroundColor;
+  ctx.lineWidth = 1;
+  roundedRect(ctx, x, y - 10, measureTextCached(typeName, ctx, tagFont).width + 16, 18, 4);
+  ctx.stroke();
+  ctx.fill();
+  ctx.fillStyle = '#fff';
```
Comment (ashtonG) [inline src/kit/DataGrid/custom-renderers/utils.ts:133]: can we use the data grid theme here? the colors should probably respond to the overall theme. i see in the theme there's a `textBubble` and `bgBubble` theme that we could use here [source](https://docs.grid.glideapps.com/api/common-types#theme).

Comment (keita-determined) [reply src/kit/DataGrid/custom-renderers/utils.ts:133]: ## light mode

image


## dark mode

image

what do you think? I think it's hard to look at the type badge. Also, should badge color be the same or similar to hew badge color? (default badge)

image

Comment (ashtonG) [reply src/kit/DataGrid/custom-renderers/utils.ts]: I agree -- looking at the theme it appears we use the `textBubble` theme variable as the border of the tags for the tag cell. That said, the badge background color does change with the theme -- could we reproduce the logic in the badge component here?

in addition, note that the badge font weight is bold. that's definitely something we want to retain.

Comment (keita-determined) [reply src/kit/DataGrid/custom-renderers/utils.ts]: ### Dark Mode

image

### Light Mode

image

File: src/kit/DataGrid/custom-renderers/utils.ts [hunk 2]
```diff
@@ -105,7 +109,35 @@ export function drawTextWithEllipsis(
...
+  x: number,
+  y: number,
+): void {
+  const tagFont = `bold 10px ${theme.fontFamily}`;
+  const bgColor = str2hsl(getComputedStyle(ctx.canvas).getPropertyValue('--theme-surface'));
+  const backgroundColor = hsl2str({
+    ...bgColor,
+    s: bgColor.s > 0 ? 70 : 0,
+  });
+  const textColor = getComputedStyle(ctx.canvas).getPropertyValue('--gdg-text-bubble');
```
Comment (ashtonG) [inline src/kit/DataGrid/custom-renderers/utils.ts:131]: i think for the text color we might need to copy the text color logic from the badge as well. as mentioned before, we're using that color variable for the tags text border [source](https://github.com/determined-ai/determined-ui/blob/377bbbcd8254f0ae35b488cccf9d591ef14022ef/src/kit/DataGrid/custom-renderers/cells/tagsCell.tsx#L62-L65)

Comment (keita-determined) [reply src/kit/DataGrid/custom-renderers/utils.ts:131]: `getComputedStyle(ctx.canvas).getPropertyValue('--gdg-text-bubble');` and `theme.textBubble` are equivalent. 
Did you mean `textDark` [code](https://github.com/determined-ai/hew/blob/377bbbcd8254f0ae35b488cccf9d591ef14022ef/src/kit/DataGrid/custom-renderers/cells/tagsCell.tsx#L68)?

with `textDark`

image

image

**Review by ashtonG — APPROVED** [review]
lgtm!

**Pull request merged**